numba/_typeof.c

```diff
@@ -309,11 +311,12 @@ compute_fingerprint(string_writer_t *w, PyObject *val)
     }
     if (PyList_Check(val)) {
         Py_ssize_t n = PyList_GET_SIZE(val);
+
         if (n == 0) {
-            PyErr_SetString(PyExc_ValueError,
-                            "cannot compute fingerprint of empty list");
-            return -1;
+            TRY(string_writer_put_char, w, OP_EMPTY_LIST);
+            return 0;
```

Comment:
What if this is kept as the original and let change https://github.com/numba/numba/blob/master/numba/_dispatcher.c#L519 to goto the object mode invocation instead of cleanup.